api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -135,4 +136,20 @@
   default Integer schemaId() {
     return null;
   }
+
+  /**
+   * Returns the location of the manifest list's encryption key metadata file location,
+   * or null if there is no key metadata.
+   */
+  default String keyMetadataLocation() {
+    return null;
+  }
+
+  /**
+   * Returns metadata about how the manifest list in the snapshot is encrypted,
+   * or null if the file is stored in plain text.
+   */
+  default ByteBuffer keyMetadata() {
+    return null;
+  }
```

Comment:
Open question (non-blocking): Should we consider adding in a `VoidKeyMetadata` or similar `null` type?

My thinking is similar to the `VoidTransform` for tables that were previously partitioned and then now are not. We use the `VoidTransform` (if I'm not mistaken) to help indicate that.

If users were to encrypt their table, and then decide to decrypt the table, would we need a way of expressing that?

Also, I'm just not a big fan of using `null` as a default (though I recognize that it's fine and valid here).